biz.aQute.bnd.maven/src/aQute/bnd/maven/lib/executions/PluginExecutions.java

```diff
@@ -0,0 +1,58 @@
+package aQute.bnd.maven.lib.executions;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.maven.model.PluginExecution;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import aQute.bnd.unmodifiable.Lists;
+
+public class PluginExecutions {
+
+	static final List<String> PACKAGING_GOALS = Lists.of("jar", "test-jar", "war");
+
+	private PluginExecutions() {}
+
+	public static String defaultClassifier(PluginExecution pluginExecution) {
+		List<String> goals = pluginExecution.getGoals();
+
+		return goals.contains("jar") ? "" : (goals.contains("test-jar") ? "tests" : "");
+	}
+
+	public static String extractClassifier(PluginExecution pluginExecution) {
+		Xpp3Dom rootConfiguration = (Xpp3Dom) pluginExecution.getConfiguration();
+
+		if (rootConfiguration == null) {
+			return defaultClassifier(pluginExecution);
+		}
+
+		Xpp3Dom classifierConfiguration = rootConfiguration.getChild("classifier");
+
+		if (classifierConfiguration == null) {
+			return defaultClassifier(pluginExecution);
+		}
+
+		return Optional.of(classifierConfiguration.getValue())
+			.filter(Objects::nonNull)
```

Comment:
This seems wrong on Optional. You use `of()` which means the value must be non-null. Also, an empty optional will not run the predicate anyway.